src/extract.ts

```diff
@@ -235,7 +235,11 @@ async function getCommitFromGitHubAPIRequest(githubToken: string): Promise<Commi
     };
 }
 
-async function getCommit(githubToken?: string): Promise<Commit> {
+async function getCommit(githubToken?: string, commitSha?: string): Promise<Commit> {
+    if (commitSha && githubToken) {
```

Comment:
Should this line be placed after the line [253-261](https://github.com/benchmark-action/github-action-benchmark/pull/93/files#diff-5a62c4cdfb37aae9bf48e8f7c76e57e9481efe48903e43c5b801ccfad5590d7aR253-R260) ? Mostly I can't see any reason to let this success faster though.

Reply:
Well, if you've provided the commit SHA, then your intention is for the action to use it. I don't see why it should even attempt to use the other options.